pkg/prometheus/server/operator.go

```diff
@@ -811,7 +808,24 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 		return err
 	}
 
+	cpf := p.GetCommonPrometheusFields()
+
+	switch serviceDiscoveryRole := *cpf.ServiceDiscoveryRole; serviceDiscoveryRole {
```

Comment:
I have separately made `cpf.ServiceDiscoveryRole` a pointer to a string instead of a string so it might not be necessary to use the `ptr` package to check the service discovery role in this switch statement.